src/cli.ts

```diff
@@ -117,10 +118,58 @@ function debugCliInfo(version: string, channel: string, env: Env, config: IConfi
   );
 }
 
+class VersionObject {
+  public cliVersion: string;
+  public pluginVersions: string[];
+  public osVersion: string;
+
+  public constructor(cliVersion: string, pluginVersions: string[], osVersion: string) {
+    this.cliVersion = cliVersion;
+    this.pluginVersions = pluginVersions;
+    this.osVersion = osVersion;
+  }
+}
+
+class SfdxMain extends Main {
+  // Function which returns Version object which includes cli version, plugin versions, OS and its version.
+  protected getIssueVersionObject(): VersionObject {
+    const cliVersion = this.config.userAgent;
```

Comment:
I'd like to see I'd want to see what's currently "cliVersion" split out into the 3 parts (cliVersion, architecture, and nodeVersion) as separate json properties

Reply:
Updated based on suggestion.